pkg/identity/github/principal.go

```diff
@@ -159,10 +161,16 @@ func WorkflowPrincipalFromIDToken(_ context.Context, token *oidc.IDToken) (ident
 		return nil, errors.New("missing run_attempt claim in ID token")
 	}
 
+	baseURL := `https://github.com/`
+
+	if claims.Enterprise != "" {
```

Comment:
@javanlacerda Is this something we can add for the new configuration?

@kommendorkapten Is this something you want enabled for the public instance or is this just for the GitHub deployment?

Reply:
Yes, I think this template `{{if .enterprise }}https://{{ .enterprise }.ghe.com{{else}}{{ .url }}{{end}}` should replace every place that the default {{url}} is used for github on identity/config.yaml. 
Also, It could be defined in a variable for avoiding replicating it into the config.

And btw, I saw that this PR was recently merged, but this modifies will not work as we merged github to use the new generic logic that is in `pkg/identity/ciprovider/principal.go`

Reply:
@haydentherapper this is only needed for our internal deployments as of now.